records_mover/records/delimited/sniff.py

```diff
@@ -1,44 +1,78 @@
 import chardet
-from .types import RecordsHints, BootstrappingRecordsHints, HintEncoding
-from .csv_streamer import stream_csv
-from .conversions import python_encoding_from_hint
+from contextlib import contextmanager
+from . import RecordsHints, BootstrappingRecordsHints
+from .csv_streamer import stream_csv, python_encoding_from_hint
 import io
 import logging
-from .types import MutableRecordsHints
-from .conversions import (
-    hint_compression_from_pandas,
-    hint_encoding_from_pandas,
-    hint_encoding_from_chardet
-)
-from typing import List, IO, Optional, TYPE_CHECKING
+import csv
+import gzip
+import bz2
+from .types import HintEncoding, HintRecordTerminator, HintQuoting, HintCompression
+from .conversions import hint_compression_from_pandas, hint_encoding_from_chardet
+import pandas
+from typing import List, IO, Optional, Iterator, NoReturn, Dict, TYPE_CHECKING
 if TYPE_CHECKING:
     from pandas.io.parsers import TextFileReader
 
 
 logger = logging.getLogger(__name__)
 
 
+@contextmanager
+def rewound_fileobj(fileobj: IO[bytes]) -> Iterator[IO[bytes]]:
+    if getattr(fileobj, 'closed', None) is not None:
+        closed = fileobj.closed
+    if closed:
+        logger.warning("Stream already closed")
+        # TODO: Am I catching this everywhere?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker